AlphaWallet/KeyManagement/EtherKeystore.swift

```diff
@@ -0,0 +1,647 @@
+        return signHash(combined, for: account)
+    }
+
+    func signMessage(_ message: Data, for account: EthereumAccount) -> Result<Data, KeystoreError> {
+        return signHash(message.sha3(.keccak256), for: account)
+    }
+
+    func signMessageBulk(_ data: [Data], for account: EthereumAccount) -> Result<[Data], KeystoreError> {
+        guard let key = getPrivateKeyForSigning(forAccount: account) else { return .failure(.accountMayNeedImportingAgainOrEnablePasscode) }
+
+        do {
+            var messageHashes = [Data]()
+            for i in 0...data.count - 1 {
+                let hash = data[i].sha3(.keccak256)
+                messageHashes.append(hash)
+            }
+            var data = try EthereumSigner().signHashes(messageHashes, withPrivateKey: key)
+            // TODO: Make it configurable, instead of overriding last byte.
+            for i in 0...data.count - 1 {
+                data[i][64] += 27
+            }
+            return .success(data)
+        } catch {
+            return .failure(KeystoreError.failedToSignMessage)
+        }
+    }
+
+    func signMessageData(_ message: Data?, for account: EthereumAccount) -> Result<Data, KeystoreError> {
+        guard let hash = message?.sha3(.keccak256) else { return .failure(KeystoreError.failedToSignMessage) }
+        guard let key = getPrivateKeyForSigning(forAccount: account) else { return .failure(.accountMayNeedImportingAgainOrEnablePasscode) }
+        do {
+            var data = try EthereumSigner().sign(hash: hash, withPrivateKey: key)
+            data[64] += 27
+            return .success(data)
+        } catch {
+            return .failure(KeystoreError.failedToSignMessage)
+        }
+    }
+
+    func signTransaction(_ transaction: UnsignedTransaction) -> Result<Data, KeystoreError> {
+        let signer: Signer
+        if transaction.server.chainID == 0 {
+            signer = HomesteadSigner()
+        } else {
+            signer = EIP155Signer(server: transaction.server)
+        }
+
+        do {
+            let hash = signer.hash(transaction: transaction)
+            guard let key = getPrivateKeyForSigning(forAccount: transaction.account) else { return .failure(.accountMayNeedImportingAgainOrEnablePasscode) }
+            let signature = try EthereumSigner().sign(hash: hash, withPrivateKey: key)
+            let (r, s, v) = signer.values(transaction: transaction, signature: signature)
+            let data = RLP.encode([
+                transaction.nonce,
+                transaction.gasPrice,
+                transaction.gasLimit,
+                transaction.to?.data ?? Data(),
+                transaction.value,
+                transaction.data,
+                v, r, s,
+            ])!
+            return .success(data)
+        } catch {
+            return .failure(.failedToSignTransaction)
+        }
+    }
+
+    //TODO should and can we check if the keychain has that entry without prompting for user-presence?
```

Comment:
Fix TODO.

Reply:
Have removed the TODO. Not necesssary anymore